cmd/server-main.go

```diff
@@ -286,7 +286,7 @@ func serverMain(ctx *cli.Context) {
 		getCert = globalTLSCerts.GetCertificate
 	}
 
-	globalHTTPServer = xhttp.NewServer([]string{globalMinioAddr}, handler, getCert)
+	globalHTTPServer = xhttp.NewServer([]string{globalMinioAddr}, criticalErrorHandler{handler}, getCert)
```

Comment:
`routers.go:globalHandlers[]` would be the right place to put this

Reply:
@krishnasrinivas While you can put it as the first handler in the slice this would be quite fragile.
Since `criticalErrorHandler` must be the first handler (top handler) to work properly. If handler code (higher in the handler hierarchy) would call `logger.CriticalIf`the `panic` would not be caught.   
So I placed the `criticalErrorHandler` on purpose. I even considered putting it into `cmd/http` but this wasn't possible due to our "wrong" package layout - See harsha's comment for details.

Reply:
ok..